ksql-engine/src/test/java/io/confluent/ksql/integration/EndToEndIntegrationTest.java

```diff
@@ -144,17 +144,21 @@ public void shouldRetainSelectedColumnsInPartitionBy() throws Exception {
     queryMetadata.getKafkaStreams().start();
 
     BlockingQueue<KeyValue<String, GenericRow>> rowQueue = queryMetadata.getRowQueue();
-    KeyValue<String, GenericRow> nextRow = rowQueue.poll();
-    if (nextRow != null) {
-      List<Object> columns = nextRow.value.getColumns();
-      assertEquals(5, columns.size());
-      String pageid = columns.get(3).toString();
-      assertEquals(5, pageid.length());
-      assertEquals("PAGE_", pageid.substring(0, 5));
-
-      String userid = columns.get(4).toString();
-      assertEquals(5, userid.length());
-      assertEquals("USER_", userid.substring(0, 5));
+    int rowsConsumed = 0;
+    while (rowsConsumed != 1) {
+      KeyValue<String, GenericRow> nextRow = rowQueue.poll();
```

Comment:
Should we have a timeout in here so that it doesn't potentially run forever? Or use `TestUtils.waitForCondition(..)` ?